packages/rockets-next-web/src/app/showroom/users/page.tsx

```diff
@@ -0,0 +1,319 @@
+
+  const { execute: createNewUser } = useQuery(createUser, false, {
+    onError: (error) => console.error(error),
+  });
+
+  const editUser = (data: FormData) =>
+    patch({
+      uri: `/user/${data.id}`,
+      body: data,
+    });
+
+  const { execute: editExistingUser } = useQuery(editUser, false, {
+    onError: (error) => console.error(error),
+  });
+
+  const deleteUser = (id: FormData["id"]) =>
+    del({
+      uri: `/user/${id}`,
+    });
+
+  const { execute: deleteExistingUser } = useQuery(deleteUser, false, {
+    onError: (error) => console.error(error),
+  });
+
+  const tableTheme = createTableStyles({
+    table: {
+      height: "100%",
+    },
+    root: {
+      display: "flex",
+      flexDirection: "column",
+      flex: 1,
+      overflow: "auto",
+    },
+    tableHeader: {
+      ...theme.typography.caption,
+      lineHeight: 1,
+      fontWeight: 500,
+      color: theme.palette.grey[500],
+    },
+    tableRow: {
+      backgroundColor: "#F9FAFB",
+      textTransform: "uppercase",
+    },
+    tableContainer: {
+      flex: 1,
+    },
+  });
+
+  const getRowDataById = useCallback(
+    (rowId: FormData["id"]) => {
+      return data.find((item: FormData) => item.id === rowId);
+    },
+    [data]
+  );
+
+  const editRow = useCallback(
+    (rowId: FormData["id"]) => {
+      if (!getRowDataById(rowId)) {
+        return;
+      }
+
+      setCurrentRow(getRowDataById(rowId) as FormData);
+      setDrawerState({ viewMode: DRAWER_VIEW_MODE.EDIT, isOpen: true });
+    },
+    [getRowDataById]
+  );
+
+  const viewRow = useCallback(
+    (rowId: FormData["id"]) => {
+      if (!getRowDataById(rowId)) {
+        return;
+      }
+
+      setCurrentRow(getRowDataById(rowId) as FormData);
+      setDrawerState({ viewMode: DRAWER_VIEW_MODE.DETAILS, isOpen: true });
+    },
+    [getRowDataById]
+  );
+
+  const deleteRow = useCallback(
+    async (rowId: FormData["id"]) => {
+      await deleteExistingUser(rowId);
+      fetchUsers();
+      resetDrawerState();
+    },
+    [deleteExistingUser, fetchUsers]
+  );
+
+  const handleFormSubmit = async (values: IChangeEvent<FormData>) => {
+    const fields = values.formData;
+
+    if (!fields) {
+      return;
+    }
+
+    if (drawerState.viewMode === DRAWER_VIEW_MODE.CREATION) {
+      await createNewUser(fields);
+    }
+
+    if (drawerState.viewMode === DRAWER_VIEW_MODE.EDIT) {
+      await editExistingUser(fields);
+    }
+
+    await fetchUsers();
+    resetDrawerState();
+  };
+
+  const resetDrawerState = () => {
+    setCurrentRow(undefined);
+    setDrawerState({ viewMode: null, isOpen: false });
+  };
+
+  const filteredRows: Record<string, string>[] = useMemo(() => {
+    if (!data) {
+      return [];
+    }
+
+    return data.filter((row: FormData) => {
+      const formattedName = row.username.toLowerCase();
+      const formattedSearch = searchTerm.toLowerCase();
+
+      return formattedName.includes(formattedSearch);
+    });
+  }, [data, searchTerm]);
```

Comment:
Don't use useMemo here, and the search should be at API, not here, so filteredRows object should not exist, all filter should be done at API

Reply:
Block removed. API integrations will be implemented in future commits.